packages/contract-wrappers/src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -35,43 +35,48 @@ import { TransactionEncoder } from '../utils/transaction_encoder';
 import { ERC721TokenWrapper } from './erc721_token_wrapper';
-import { ExchangeContract, ExchangeEventArgs, ExchangeEvents } from './generated/exchange';
 
 /**
  * This class includes all the functionality related to calling methods, sending transactions and subscribing to
  * events of the 0x V2 Exchange smart contract.
  */
 export class ExchangeWrapper extends ContractWrapper {
-    public abi: ContractAbi = artifacts.Exchange.compilerOutput.abi;
+    public abi: ContractAbi = Exchange.compilerOutput.abi;
+    public address: string;
+    public zrxTokenAddress: string;
     private _exchangeContractIfExists?: ExchangeContract;
     private _erc721TokenWrapper: ERC721TokenWrapper;
     private _erc20TokenWrapper: ERC20TokenWrapper;
-    private _contractAddressIfExists?: string;
-    private _zrxContractAddressIfExists?: string;
     /**
      * Instantiate ExchangeWrapper
-     * @param web3Wrapper Web3Wrapper instance to use
-     * @param networkId Desired networkId
-     * @param contractAddressIfExists The exchange contract address to use. This is usually pulled from
-     * the artifacts but needs to be specified when using with your own custom testnet.
-     * @param zrxContractAddressIfExists The ZRXToken contract address to use. This is usually pulled from
-     * the artifacts but needs to be specified when using with your own custom testnet.
-     * @param blockPollingIntervalMs The block polling interval to use for active subscriptions
+     * @param web3Wrapper Web3Wrapper instance to use.
+     * @param networkId Desired networkId.
+     * @param erc20TokenWrapper ERC20TokenWrapper instance to use.
+     * @param erc721TokenWrapper ERC721TokenWrapper instance to use.
+     * @param address (Optional) The address of the Exchange contract. If
+     * undefined, will default to the known address corresponding to the
+     * networkId.
+     * @param zrxTokenAddress (Optional) The address of the ZRXToken contract.
+     * If undefined, will default to the known address corresponding to the
+     * networkId.
+     * @param blockPollingIntervalMs The block polling interval to use for active subscriptions.
      */
     constructor(
         web3Wrapper: Web3Wrapper,
         networkId: number,
         erc20TokenWrapper: ERC20TokenWrapper,
         erc721TokenWrapper: ERC721TokenWrapper,
-        contractAddressIfExists?: string,
-        zrxContractAddressIfExists?: string,
+        address?: string,
+        zrxTokenAddress?: string,
         blockPollingIntervalMs?: number,
     ) {
         super(web3Wrapper, networkId, blockPollingIntervalMs);
         this._erc20TokenWrapper = erc20TokenWrapper;
         this._erc721TokenWrapper = erc721TokenWrapper;
-        this._contractAddressIfExists = contractAddressIfExists;
-        this._zrxContractAddressIfExists = zrxContractAddressIfExists;
+        this.address = _.isUndefined(address) ? this._getDefaultContractAddresses().exchange : address;
```

Comment:
We need a more intelligible error message here too. Thinking about this again, what we should really do is **assert** that if `networkId` isn't one of our supported addresses, that `address` **is** supplied.